File: data_structures/trees/depth_first_search/java/depthFirstSearch.java
class depthFirstSearch
{
    public static boolean search(Node curr, int val)
    {
        if(curr == null)
        {
            return false;
        }
        if(curr.val == val)
        {
            return true;
        }
        if(val > curr.val)
        {
            return search(curr.right, val);
        }
        return search(curr.left, val);
    }
    public static void main(String[] args) {
        Node bt_head = new Node(25);
        Node left = new Node(23);
        left.left = new Node(5);
        Node right = new Node(27);
        bt_head.left = left;
        bt_head.right = right;
        right.left = new Node(26);
        right.right = new Node(30);
        right.right.right = new Node(100);

        System.out.println(search(bt_head, 5)); //true

        System.out.println(search(bt_head, 23)); //true

        System.out.println(search(bt_head, 30)); //true

        System.out.println(search(bt_head, 0)); //false

        System.out.println(search(bt_head, 28)); //false

        System.out.println(search(bt_head, 2356)); //false
    }
}

File: data_structures/trees/depth_first_search/python/depthFirstSearch.py
class Node():
    def __init__(self, val, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
    
def depthFirstSearch(curr, val):
    """Depth First Search"""

    if (curr == None):
        return False
    if (curr.val == val):
        return True
    elif (val > curr.val):
        return depthFirstSearch(curr.right, val)
    elif (val <= curr.val):
        return depthFirstSearch(curr.left, val)

if __name__ == "__main__":
    bt_head = Node(25, 
                   left=Node(23, Node(5, Node(2), None)),
                   right=Node(27, 
                              Node(26), Node(30, left=None, right=Node(100))
                              )
                    )

    print(depthFirstSearch(bt_head, 5)) #true

    print(depthFirstSearch(bt_head, 23)) #true

    print(depthFirstSearch(bt_head, 30)) #true

    print(depthFirstSearch(bt_head, 0)) #false

    print(depthFirstSearch(bt_head, 28)) #false

    print(depthFirstSearch(bt_head, 2356)) #false


